DynamoDB Persistence Service has build problem

Hi @ssalonen,

I believe this repo is being built for Java 8 now (https://github.com/openhab/openhab1-addons/commit/2a1b7e1f88eaf313fa7d48b36edce0ac97fcc491), and this is a new build error:

```log
12:25:49.663  |  [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.24.0:compile (default-compile) on project org.openhab.persistence.dynamodb: Compilation failure: Compilation failure:
12:25:49.663  |  [ERROR] /scratch/jenkins/workspace/openHAB1-Addons/bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java:[323]
12:25:49.663  |  [ERROR] paginatedList = mapper.query(dtoClass, queryExpression);
12:25:49.663  |  [ERROR] ^^^^^
12:25:49.664  |  [ERROR] The method query(Class<T>, DynamoDBQueryExpression<T>) in the type AbstractDynamoDBMapper is not applicable for the arguments (Class<capture#19-of ? extends DynamoDBItem<?>>, DynamoDBQueryExpression<DynamoDBItem<?>>)
12:25:49.664  |  [ERROR] 1 problem (1 error)
12:25:49.664  |  [ERROR] -> [Help 1]
12:25:49.664  |  [ERROR] 
12:25:49.664  |  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
12:25:49.664  |  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
12:25:49.664  |  [ERROR] 
12:25:49.665  |  [ERROR] For more information about the errors and possible solutions, please read the following articles:
12:25:49.665  |  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
12:25:49.665  |  [ERROR] 
12:25:49.665  |  [ERROR] After correcting the problems, you can resume the build with the command
12:25:49.665  |  [ERROR]   mvn <goals> -rf :org.openhab.persistence.dynamodb
```

Would you have an opportunity to have a look?

See also #5028 and https://github.com/openhab/openhab1-addons/commit/f7787764b93748c2e8465a697b7f0688c424acb8

--- a/bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java
+++ b/bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java
@@ -78,9 +78,10 @@ public abstract class AbstractDynamoDBItem<T> implements DynamoDBItem<T> {
         itemClassToDynamoItemClass.put(ColorItem.class, DynamoDBStringItem.class); // inherited from DimmerItem
     }
 
-    public static final Class<? extends DynamoDBItem<?>> getDynamoItemClass(Class<? extends Item> itemClass)
+    public static final Class<DynamoDBItem<?>> getDynamoItemClass(Class<? extends Item> itemClass)
             throws NullPointerException {
-        Class<? extends DynamoDBItem<?>> dtoclass = itemClassToDynamoItemClass.get(itemClass);
+        @SuppressWarnings("unchecked")
+        Class<DynamoDBItem<?>> dtoclass = (Class<DynamoDBItem<?>>) itemClassToDynamoItemClass.get(itemClass);
         if (dtoclass == null) {
             throw new NullPointerException(String.format("Unknown item class %s", itemClass));
         }

--- a/bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java
+++ b/bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java
@@ -308,7 +308,7 @@ public class DynamoDBPersistenceService implements QueryablePersistenceService {
             return Collections.emptyList();
         }
 
-        Class<? extends DynamoDBItem<?>> dtoClass = AbstractDynamoDBItem.getDynamoItemClass(item.getClass());
+        Class<DynamoDBItem<?>> dtoClass = AbstractDynamoDBItem.getDynamoItemClass(item.getClass());
         String tableName = tableNameResolver.fromClass(dtoClass);
         DynamoDBMapper mapper = getDBMapper(tableName);
         logger.debug("item {} (class {}) will be tried to query using dto class {} from table {}", itemName,

--- a/bundles/persistence/pom.xml
+++ b/bundles/persistence/pom.xml
@@ -19,8 +19,8 @@
     <module>org.openhab.persistence.caldav</module>    
     <module>org.openhab.persistence.db4o</module>
     <module>org.openhab.persistence.db4o.test</module>
-<!--    <module>org.openhab.persistence.dynamodb</module>
-    <module>org.openhab.persistence.dynamodb.test</module>-->
+    <module>org.openhab.persistence.dynamodb</module>
+    <module>org.openhab.persistence.dynamodb.test</module>
     <module>org.openhab.persistence.logging</module>
     <module>org.openhab.persistence.sense</module>
     <module>org.openhab.persistence.rrd4j</module>

--- a/distribution/pom.xml
+++ b/distribution/pom.xml
@@ -1226,13 +1226,13 @@
       <version>${project.version}</version>
       <type>jar</type>
     </dependency>
-<!--    <dependency>
+    <dependency>
       <groupId>org.openhab.persistence</groupId>
       <artifactId>org.openhab.persistence.dynamodb</artifactId>
       <version>${project.version}</version>
       <type>jar</type>
     </dependency>
--->    <dependency>
+    <dependency>
       <groupId>org.openhab.persistence</groupId>
       <artifactId>org.openhab.persistence.exec</artifactId>
       <version>${project.version}</version>

